static/app.js

```diff
@@ -262,4 +278,40 @@
 		}
 	})
 
+	app.controller('areaHealthModificationCtrl', function($scope, $modalInstance, characters){
+		$scope.areaDamage = 0;
+
+		$scope.characters = characters;
+		$scope.characters.forEach(function (character) {
+			character.damaged = false;
+			character.damageHalved = false;
+		});
+
+		$scope.toggleDamaged = function (character) {
```

Comment:
If you tick damaged and damageHalved, and after you remove damaged flag, damageHalved remains ticked.